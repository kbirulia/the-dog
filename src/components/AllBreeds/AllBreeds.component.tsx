import React from "react";
import IBreed from "../../interfaces/breed";
import "./AllBreeds.style.css";
import Spinner from "../Spinner";
import IErrorStatus from "../../interfaces/IErrorStatus";

interface IAllBreedProps {
  error: null | IErrorStatus;
  isLoaded: boolean;
  breeds: ReadonlyArray<IBreed>;
}

const AllBreeds = ({
  error,
  isLoaded,
  breeds,
}: IAllBreedProps): JSX.Element => {
  if (error) {
    return <div>Error: {error.statusText}</div>;
  }
  if (!isLoaded) {
    return <Spinner />;
  }
  return (
    <>
      <h2 className="main__title">All breeds</h2>
      <ul>
        {breeds.map((breed: IBreed) => (
          <li key={breed.id}>{breed.name}</li>
        ))}
      </ul>
    </>
  );
};

export default AllBreeds;
