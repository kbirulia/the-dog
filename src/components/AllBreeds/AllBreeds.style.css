.main__title {
  align-self: center;
}
