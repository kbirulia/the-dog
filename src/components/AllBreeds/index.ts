export { default } from "./AllBreeds.container";
