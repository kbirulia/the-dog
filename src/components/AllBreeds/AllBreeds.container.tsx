import React from "react";
import useFetch from "../../hooks/useFetch";
import AllBreeds from "./AllBreeds.component";
import IBreed from "../../interfaces/breed";
import FetchResults from "../FetchResults";

const AllBreedsContainer = (): JSX.Element => {
  const { error, isLoaded, data } = useFetch<ReadonlyArray<Partial<IBreed>>>(
    [],
    "https://api.thedogapi.com/v1/breeds"
  );

  return (
    <FetchResults error={error} isLoaded={isLoaded}>
      <AllBreeds breeds={data} />
    </FetchResults>
  );
};

export default AllBreedsContainer;
