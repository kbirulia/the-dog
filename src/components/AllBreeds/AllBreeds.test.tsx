import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import AllBreeds from "./AllBreeds.component";

describe("AllBreeds", () => {
  test("should render breeds", () => {
    const tree = renderer
      .create(
        <Router>
          <AllBreeds
            breeds={[
              {
                id: 1,
                name: "African Hunting Do",
              },
              {
                id: 2,
                name: "Airedale Terrier",
              },
            ]}
          />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
