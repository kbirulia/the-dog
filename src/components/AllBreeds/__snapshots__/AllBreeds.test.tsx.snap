// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllBreeds should render breeds 1`] = `
Array [
  <h2
    className="main__title"
  >
    All breeds
  </h2>,
  <ul>
    <li>
      <a
        href="/breeds/1"
        onClick={[Function]}
      >
        African Hunting Do
      </a>
    </li>
    <li>
      <a
        href="/breeds/2"
        onClick={[Function]}
      >
        Airedale Terrier
      </a>
    </li>
  </ul>,
]
`;
